packages/metro/src/lib/contextModuleTemplates.js

```diff
@@ -0,0 +1,138 @@
+import * as path from 'path';
+import type {
+  ContextMode,
+} from '../ModuleGraph/worker/collectDependencies';
+
+function createFileMap(
+    modulePath: string,
+    files: string[],
+    processModule: (moduleId: string) => string,
+) {
+    let mapString = '';
+
+    files.map(file => {
+        let filePath = path.relative(modulePath, file);
+
+        // NOTE(EvanBacon): I'd prefer we prevent the ability for a module to require itself (`require.context('./')`)
+        // but Webpack allows this, keeping it here provides better parity between bundlers.
+
+        // Ensure relative file paths start with `./` so they match the
+        // patterns (filters) used to include them.
+        if (!filePath.startsWith('.')) {
+        filePath = `.${path.sep}` + filePath;
+        }
+        const key = JSON.stringify(filePath);
+        // NOTE(EvanBacon): Webpack uses `require.resolve` in order to load modules on demand,
+        // Metro doesn't have this functionality so it will use getters instead. Modules need to
+        // be loaded on demand because if we imported directly then users would get errors from importing
+        // a file without exports as soon as they create a new file and the context module is updated.
```

Comment:
> Errors from importing a file without exports

What errors? Are these built into Metro or are they a result of application logic?

Does Webpack not eagerly evaluate the inner `require`s as part of the `require.context()` call?

Reply:
> Does Webpack not eagerly evaluate the inner requires as part of the require.context() call?

No. The context module is seemingly always loaded on demand with `ctx(moduleId)`, the different `mode`s change how the module will be imported when requested (promise, load promise once, require, etc.).

> ...

My comment here is in regards to the edge case that occurs when a file is first created, the module will attempt to import a module with no exports, resulting in an error.